### PR TITLE
Ar join event form

### PR DESCRIPTION
**Pull request by arthur-ruxton**

### CODE REVIEW

File: src/pages/EventPage.jsx [hunk 1]
```diff
@@ -1,14 +1,51 @@
...
+    get_event_details(event_id).then(details => setEventDetails(details));
+  }, [event_id]);
+
   return (<>
-    Event is {event_id}
-    <JoinEventForm event_id={event_id} hide_event_id={true}/>
+    {eventDetails && <>
+      Event name is {eventDetails.name} taking place at lat: {eventDetails.lat}, lon: {eventDetails.lon}<br/>
+    </>}
+    {hasJoined ||
```
Comment (arthur-ruxton) [inline src/pages/EventPage.jsx:43]: I think this may need to be adjusted to use a ternary operator:  {hasJoined ? hasJoined : <JoinEventFrom .../>}

Comment (benkrikler) [reply src/pages/EventPage.jsx:43]: Thanks for checking, @arthur-ruxton! The ternary operator here, if used in that way, should just give me the same result as what I have now i believe, due to the short-circuit evaluation of the `||` operator. So the right hand side of `||` is only evaluated if and only if the left hand side evaluates to false. Unless there's a style convention I'm not aware of that you're referring to maybe?

Comment (arthur-ruxton) [reply src/pages/EventPage.jsx:43]: AH makes sense, didn't know we could do that, very cool.

**Comment by benkrikler** [comment]
This PR should be complete from my side now - in particular the websocket is hooked up in a first demo component. Next steps in a subsequent PR should display the results better and add overall styling.  Will wait a day or so for reviews of this but if I don't get anything back will merge ahead anyway to move quickly :slightly_smiling_face: 

Note this needs the version of the API as implemented on https://github.com/RemotelyGreenOrg/co2-calculator-api/pull/28 to function.

**Pull request merged**